packages/server/src/sdk/app/rows/search/sqs.ts

```diff
@@ -90,6 +92,33 @@ function cleanupFilters(
       )
   )
 
+  // generate a map of all possible column names (these can be duplicated across tables
+  // the map of them will always be the same
+  const userColumnMap: Record<string, string> = {}
+  allTables.forEach(table =>
+    Object.keys(table.schema).forEach(
+      key => (userColumnMap[key] = mapToUserColumn(key))
+    )
+  )
+
+  // sort longest first, don't find substrings
+  const userColumnList = Object.keys(userColumnMap).sort(
+    (a, b) => b.length - a.length
+  )
+  // update the keys of filters to manage user columns
+  for (let filter of Object.values(filters)) {
+    for (let key of Object.keys(filter)) {
+      const found = userColumnList.find(possibleColumn =>
+        key.endsWith(possibleColumn)
+      )
+      if (found) {
+        const newKey = key.replace(found, userColumnMap[found])
+        filter[newKey] = filter[key]
+        delete filter[key]
+      }
+    }
+  }
```

Comment:
Could we not simplify this section quite a lot?

```typescript
function keyInAnyTable(key: string): boolean {
  return allTables.some(table => table.schema[key])
}

for (const filter of Object.values(filters)) {
  for (const key of Object.keys(filter)) {
    const [prefix, rawKey] = splitKeyAndNumericPrefix(key) // don't think this exists yet we'd have to write it
    if (keyInAnyTable(rawKey)) {
      filter[`${prefix}${mapToUserColumn(rawKey)}`] = filter[key]
      delete filter[key]
    }
  }
}
```

We also remove a bunch of array scans and any possibility of doing an incorrect replacement.

Reply:
I've written a new function which requires metadata from the tables to work this out instead.